apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -912,14 +928,22 @@ block_txs_count_pending(_Config) ->
                     get_block_txs_count_preset("pending"),
     rpc(application, set_env, [aecore, expected_mine_rate,
                                60 * 60 * 1000]), % aim at one block an hour
+    InsertedTxsCount = add_spend_txs(),
     rpc(aec_conductor, start_mining, []),
-    rpc(aec_conductor, start_mining, []),
-    timer:sleep(100),% so the miner is started
-    add_spend_txs(),
-    {ok, PendingBlock} = get_pending_block(),
-    TxsCount = length(aec_blocks:txs(PendingBlock)), 
-    {ok, 200, #{<<"count">> := TxsCount}} =
-        get_block_txs_count_preset("pending"),
+    GetPending =
+        fun TryToGet(Attempts) when Attempts < 1 ->
+                {error, no_pending_block_returned};
+            TryToGet(Attempts) ->
+                case get_block_txs_count_preset("pending") of
+                    {ok, 200, P} -> {ok, P};
+                    {ok, 404, _} ->
+                        timer:sleep(100),
+                        TryToGet(Attempts -1)
+                end
+        end,
+    ct:log("Inserted transactions count ~p", [InsertedTxsCount]),
+    {ok, #{<<"count">> := TxsCount}} = GetPending(10),
+    true = TxsCount =:= InsertedTxsCount + 1,
```

Comment:
On failure this line will say "badmatch of value false on line xxx". May I suggest making potential failure more informative?
* Either log TxsCount (the test already logs InsertedTxsCount.); or
* Replace line with `{TxsCount, TxsCount} = {(InsertedTxsCount + 1), TxsCount}` (or similar).

Reply:
Actually we do have the TxsCount logged as we log every request being made
```

*** User 2018-01-16 14:46:36.735 ***
GET "http://127.0.0.1:3113/v1/block/txs/count/pending"


*** User 2018-01-16 14:46:36.736 ***
Return code 200, Body "{\"count\":5}"
```
I found it redundant to log it once more

Reply:
Ah I see. Ok then.

Reply:
I've added a log for better clarity